src/rendezvous/transports/MSC3886SimpleHttpRendezvousTransport.ts

```diff
@@ -0,0 +1,197 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { UnstableValue } from 'matrix-events-sdk';
+
+import { logger } from '../../logger';
+import { sleep } from '../../utils';
+import {
+    RendezvousFailureListener,
+    RendezvousFailureReason,
+    RendezvousTransport,
+    RendezvousTransportDetails,
+} from '..';
+import { MatrixClient } from '../../matrix';
+import { ClientPrefix } from '../../http-api';
+
+const TYPE = new UnstableValue("http.v1", "org.matrix.msc3886.http.v1");
+
+export interface MSC3886SimpleHttpRendezvousTransportDetails extends RendezvousTransportDetails {
+    uri: string;
+}
+
+/**
+ * Implementation of the unstable [MSC3886](https://github.com/matrix-org/matrix-spec-proposals/pull/3886)
+ * simple HTTP rendezvous protocol.
+ * Note that this is UNSTABLE and may have breaking changes without notice.
+ */
+export class MSC3886SimpleHttpRendezvousTransport implements RendezvousTransport {
+    private uri?: string;
+    private etag?: string;
+    private expiresAt?: Date;
+    private client: MatrixClient;
+    private fallbackRzServer?: string;
+    private fetchFn?: typeof global.fetch;
+    private cancelled = false;
+    private _ready = false;
+    public onFailure?: RendezvousFailureListener;
+
+    public constructor({
+        onFailure,
+        client,
+        fallbackRzServer,
+        fetchFn,
+    }: {
+        fetchFn?: typeof global.fetch;
+        onFailure?: RendezvousFailureListener;
+        client: MatrixClient;
+        fallbackRzServer?: string;
+    }) {
+        this.fetchFn = fetchFn;
+        this.onFailure = onFailure;
+        this.client = client;
+        this.fallbackRzServer = fallbackRzServer;
+    }
+
+    get ready() {
+        return this._ready;
+    }
+
+    public async details(): Promise<MSC3886SimpleHttpRendezvousTransportDetails> {
+        if (!this.uri) {
+            throw new Error('Rendezvous not set up');
+        }
+
+        return {
+            type: TYPE.name,
+            uri: this.uri,
+        };
+    }
+
+    private fetch(resource: URL | string, options?: RequestInit): ReturnType<typeof global.fetch> {
+        if (this.fetchFn) {
+            return this.fetchFn(resource, options);
+        }
+        return global.fetch(resource, options);
+    }
+
+    private async getPostEndpoint(): Promise<string | undefined> {
+        try {
+            if (await this.client.doesServerSupportUnstableFeature('org.matrix.msc3886')) {
+                return `${this.client.baseUrl}${ClientPrefix.Unstable}/org.matrix.msc3886/rendezvous`;
+            }
+        } catch (err) {
+            logger.warn('Failed to get unstable features', err);
+        }
+
+        return this.fallbackRzServer;
+    }
+
+    public async send(data: object) {
```

Comment:
Missing return type, lacking types, we should avoid things like `any` and `object` - preferring Generics